From b436254cd55dc0d2ebb6d2ca7ce5f58520ea5dbc Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Thu, 24 Sep 2015 18:59:14 +0800
Subject: [PATCH] MLK-11623 ASoC: imx-cs42888: add 32k and 64k sample rate
 support

When codec sysclk is 24576000, the sample rate ratio can be 128, 192,
256, 384, 512, 768, 1024. So 32k, 48k, 64k, 96k, 192k can be support.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
---
 sound/soc/fsl/imx-cs42888.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/sound/soc/fsl/imx-cs42888.c b/sound/soc/fsl/imx-cs42888.c
index 7b0047a..26e0809 100644
--- a/sound/soc/fsl/imx-cs42888.c
+++ b/sound/soc/fsl/imx-cs42888.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2010-2014 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2010-2015 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -80,11 +80,13 @@ static int imx_cs42888_surround_startup(struct snd_pcm_substream *substream)
 	int ret;
 
 	if (priv->mclk_freq == 24576000) {
-		support_rates[0] = 48000;
-		support_rates[1] = 96000;
-		support_rates[2] = 192000;
+		support_rates[0] = 32000;
+		support_rates[1] = 48000;
+		support_rates[2] = 64000;
+		support_rates[3] = 96000;
+		support_rates[4] = 192000;
 		constraint_rates.list = support_rates;
-		constraint_rates.count = 3;
+		constraint_rates.count = 5;
 
 		ret = snd_pcm_hw_constraint_list(runtime, 0, SNDRV_PCM_HW_PARAM_RATE,
 							&constraint_rates);
-- 
1.8.0

